src/types/stardust/block/payload/transaction.rs

```diff
@@ -104,17 +112,24 @@ impl From<TransactionPayload> for bee::dto::TransactionPayloadDto {
     }
 }
 
+/// Represents the essence of a transaction.
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(rename_all = "snake_case", tag = "kind")]
 pub enum TransactionEssence {
+    /// The regular transaction essence.
     Regular {
+        /// Including the network id in the transaction prevents replay attacks.
         #[serde(with = "crate::types::util::stringify")]
         network_id: u64,
+        /// The list of inputs that this transaction consumes.
         inputs: Box<[Input]>,
         #[serde(with = "bytify")]
+        /// The input commitment.
```

Comment:
I think this is a hash over the input ids.... so "The input commitment hash as bytes" might be a better (because more concrete) doc.